src/app/games/apagos/ApagosFailure.ts

```diff
@@ -0,0 +1,12 @@
+export class ApagosFailure {
+
+    public static readonly PIECE_SHOULD_MOVE_DOWNWARD: () => string = () => $localize`Pieces should only move downward!`;
+
+    public static readonly CANNOT_LAND_ON_A_FULL_SQUARE: () => string = () => $localize`That square is already full, you cannot put a piece in it!`;
+
+    public static readonly NO_PIECE_OF_YOU_IN_CHOSEN_SQUARE: () => string = () => $localize`You have no remaining piece in that square, select one that contains at least one of your pieces!`;
```

Comment:
pieces*

Reply:
tiens d'ailleurs, ce failure est bizarre quand il s'affiche, car parfois y'a jamais eu de pièce dans le carré en question.
Peut-être simplement "You have no pieces in that square" ?

Reply:
Fait!